src/command_executor.cc

```diff
@@ -32,9 +47,19 @@ int CommandExecutor::ProcessChild(const std::string &command,
     utils::PrintSystemError(std::cerr);
   }
 
-  auto argv = BuildArgv(command, args);
+  auto argv = BuildArgv(redirectElement.command, redirectElement.args);
+  if (redirectElement.cmd_type == redirectElement.cmd_type::direct) {
+    if (!redirectElement.input_redirect_file.empty()) {
```

Comment:
You already check `input_redirect_file` and `output_redirect_file` here, so do we really need `redirectElement.cmd_type ` ?